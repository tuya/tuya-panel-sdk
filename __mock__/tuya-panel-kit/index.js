import React from 'react';
import { View } from 'react-native';
import TYText from './components/TYText';
import UnitText from './components/UnitText';

module.exports = {
  TYText,
  UnitText,
  Utils: {
    RatioUtils: {
      convertX: () => {},
    },
  },
};
