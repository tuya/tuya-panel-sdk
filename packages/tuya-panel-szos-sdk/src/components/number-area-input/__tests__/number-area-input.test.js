import React, { useRef } from 'react';
import { shallow } from 'enzyme';
import NumerAreaInput from '../index';
import { GlobalToast } from 'tuya-panel-kit';

describe('NumerAreaInput components', () => {
  it('basic render', () => {
    const wrapper = shallow(<NumerAreaInput name="test" key="test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render focus', () => {
    const changeColor = false;
    const maxVal = 30;
    const func = () => {
      GlobalToast.show({
        text: 'focus',
        onFinish: () => {
          GlobalToast.hide();
        },
      });
    };
    const wrapper = shallow(
      <NumerAreaInput
        key="test"
        name="test"
        placeholder="6"
        focusFuc={func}
        minVal={5}
        maxVal={maxVal}
        editable={true}
        changeColor={changeColor}
        viewStyle={{ borderColor: 'red', borderWidth: 1, borderRadius: 5, marginTop: 20 }}
      />
    );
    const wrapperHandle = wrapper.children().at(0);
    wrapperHandle.simulate('focus');
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  it('render change', () => {
    const changeText = val => {
      GlobalToast.show({
        text: val,
        onFinish: () => {
          GlobalToast.hide();
        },
      });
    };
    const wrapper = shallow(
      <NumerAreaInput key="test" name="test" minVal="5" maxVal="10" changeText={changeText} />
    );
    const wrapperHandle = wrapper.children().at(0);
    wrapperHandle.simulate('changeText', '0');
    wrapperHandle.simulate('changeText', '5');
    wrapperHandle.simulate('changeText', '6');
    wrapperHandle.simulate('changeText', '20');
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
