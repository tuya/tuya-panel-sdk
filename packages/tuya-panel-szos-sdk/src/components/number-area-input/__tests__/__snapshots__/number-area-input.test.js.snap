// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumerAreaInput components basic render 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
      },
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "lineHeight": 60,
        "width": 100,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    clearTextOnFocus={true}
    editable={true}
    key="test"
    keyboardType="numeric"
    maxLength={3}
    multiline={true}
    numberOfLines={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder=""
    placeholderTextColor="rgba(103, 112, 123, 0.5)"
    rejectResponderTermination={true}
    selectionColor="rgba(152, 165, 198, 1)"
    style={
      Array [
        Object {
          "alignItems": "center",
          "color": "rgba(103, 112, 123, 1)",
          "fontSize": 30,
          "height": 30,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": "100%",
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`NumerAreaInput components render change 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
      },
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "lineHeight": 60,
        "width": 100,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    clearTextOnFocus={true}
    editable={true}
    key="test"
    keyboardType="numeric"
    maxLength={3}
    multiline={true}
    numberOfLines={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder=""
    placeholderTextColor="rgba(103, 112, 123, 0.5)"
    rejectResponderTermination={true}
    selectionColor="rgba(152, 165, 198, 1)"
    style={
      Array [
        Object {
          "alignItems": "center",
          "color": "rgba(103, 112, 123, 1)",
          "fontSize": 30,
          "height": 30,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": "100%",
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`NumerAreaInput components render focus 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "red",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginTop": 20,
      },
      Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "lineHeight": 60,
        "width": 100,
      },
    ]
  }
>
  <TextInput
    allowFontScaling={true}
    clearTextOnFocus={true}
    editable={true}
    key="test"
    keyboardType="numeric"
    maxLength={3}
    multiline={true}
    numberOfLines={1}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder=""
    placeholderTextColor="rgba(103, 112, 123, 0.5)"
    rejectResponderTermination={true}
    selectionColor="rgba(152, 165, 198, 1)"
    style={
      Array [
        Object {
          "alignItems": "center",
          "color": "rgba(103, 112, 123, 1)",
          "fontSize": 30,
          "height": 30,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
          "textAlign": "center",
          "textAlignVertical": "center",
          "width": "100%",
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
